src/TritonRoute/include/triton_route/TritonRoute.h

```diff
@@ -72,6 +72,21 @@ class TritonRoute
   void reportConstraints();
 
   void readParams(const std::string& fileName);
+  void setParams(const std::string& guide_file,
+                 const std::string& output_guide_file,
+                 const std::string& output_maze_file,
+                 const std::string& output_DRC_file,
+                 const std::string& output_CMap_file,
+                 const std::string& dbProcessNode,
+                 int drouteEndIterNum,
+                 int drouteViaInPinBottomLayerNum,
+                 int drouteViaInPinTopLayerNum,
+                 int or_seed,
+                 double or_k,
+                 int bottomRoutingLayer,
+                 int topRoutingLayer,
+                 int initRouteShapeCost,
+                 int verbose);
```

Comment:
@maliberty is it preferable to have a function with a large argument list or individual setters for each value?

Reply:
I tend to prefer a to pass a struct.  It can also be helpful to capture the default values, something like:

struct Params
{
    int a = 5;
    const char* b = "abc";
};

Reply:
But when calling it from TCL the interface function will still have many arguments, right? Is there a way to pass a struct from TCL?

Reply:
I should be possible to do with swig though it might be a pain.  The other choice is to handle the conversion in the .i file so at least we stop the propagation of the big list.